embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/comms/delivery/EmbraceDeliveryService.kt

```diff
@@ -88,59 +84,33 @@ internal class EmbraceDeliveryService(
         apiService.saveLogEnvelope(logEnvelope)
     }
 
-    override fun sendCrash(crash: EventMessage, processTerminating: Boolean) {
-        runCatching {
-            cacheManager.saveCrash(crash)
-            val future = apiService.sendCrash(crash)
-
-            if (processTerminating) {
-                future.get(CRASH_TIMEOUT, TimeUnit.SECONDS)
-            }
-        }
-    }
-
     override fun sendCachedSessions(
         nativeCrashServiceProvider: Provider<NativeCrashService?>,
         sessionIdTracker: SessionIdTracker
     ) {
-        sendCachedCrash()
-        backgroundWorker.submit(TaskPriority.HIGH) {
```

Comment:
is it okay to stop using the backgroundWorker for this?

Reply:
It would not be, this still needs to run on a background thread. I've refactored it so `EmbraceDeliveryService` doesn't know anything about the worker: https://github.com/embrace-io/embrace-android-sdk/pull/1311/files#diff-c270387adce360d4187ce687c82e19785f634d47e6989038139d343ac00da9b4R207

Reply:
awesome, I hadn't reached that part yet 😅